### PR TITLE
Expand installation notes & update good-first-issues link

### PR DESCRIPTION
**Pull request by viniciusdc**
Includes a fix note for handling the `Error getting the version from source "vcs"` error for first-time contributors and addresses the TODO point for to linking the good first time issues page on GitHub

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [x] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [x] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [x] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

File: docs/docs/community/code-contributions.mdx [hunk 1]
```diff
@@ -16,10 +16,8 @@ To contribute to the documentation, read [Contribute to Nebari's documentation][
 
 ## Select an issue to work on
 
-<!-- TODO: Add link to "good first issue" labels after the qhub -> Nebari rename -->
-
 New to Nebari?
-The issues marked with the "good first issue" label are an excellent place to start.
+The issues marked with the ["good first issue"][good-first-issues] label are an excellent place to start.
```
Comment (viniciusdc) [inline docs/docs/community/code-contributions.mdx:20]:
```suggestion
The issues marked with the ["good first issue"][good-first-issue] label are an excellent place to start.
```

File: docs/docs/community/code-contributions.mdx [hunk 2]
```diff
@@ -248,6 +250,7 @@ Approval means the core team member has carefully reviewed the changes, and the
 [nebari-repo]: https://github.com/nebari-dev/nebari
 [nebari-docs-repo]: https://github.com/nebari-dev/nebari-docs
 [nebari-docs]: https://www.nebari.dev/
+[good-first-issues]: https://github.com/nebari-dev/nebari/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22
```
Comment (viniciusdc) [inline docs/docs/community/code-contributions.mdx:253]:
```suggestion
[good-first-issue]: https://github.com/nebari-dev/nebari/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22
```

File: docs/docs/community/code-contributions.mdx [hunk 3]
```diff
@@ -110,6 +108,10 @@ python -m pip install -e .\[dev\]
 ```
 :::
 
+:::note
+If you encounter the following issue while attempting to install the development version of Nebari `AssertionError: Error getting the version from source `vcs``, make sure that all tags were also pulled and synced with your forked git repository (`git fetch --tags upstream`), as Nebari heavily depends on those tags as a source to build its development and production versions dynamically.
```
Comment (pavithraes) [inline docs/docs/community/code-contributions.mdx:112]:
```suggestion
If you encounter the following issue while attempting to install the development version of Nebari `AssertionError: Error getting the version from source `vcs`, make sure that all tags were also pulled and synced with your forked git repository (`git fetch --tags upstream`), as Nebari heavily depends on those tags as a source to build its development and production versions dynamically.  
```

**Review by pavithraes — APPROVED** [review]
Thank you, Vini! LGTM, only a typo which I'll fix-up and merge. 🌻

**Pull request merged**